src/ode_solve.jl

```diff
@@ -286,6 +286,21 @@ function generate_loss(strategy::StochasticTraining, phi, f, autodiff::Bool, tsp
     optf = OptimizationFunction(loss, Optimization.AutoZygote())
 end
 
+function generate_loss(strategy::WeightedGridTraining, phi, f, autodiff::Bool, tspan, p, batch)
+    ts = tspan[1]:(strategy.dx):tspan[2]
```

Comment:
Even with WeightedGridTraining, the loss computation is still the same as GridTraining as timepoints are the same. 
So, time points for this strategy should be computed and passed here instead of in the  __solve function

Reply:
Yep that sounds like a great idea, I will work on that now